tests/components/http/test_static.py

```diff
@@ -0,0 +1,133 @@
+"""Tests for hosting static resource."""
+import os
+from tempfile import TemporaryDirectory
+
+import pytest
+
+from homeassistant.setup import async_setup_component
+
+
+async def test_register_static_path(hass, hass_client, aiohttp_client):
+    """Test register a dir to host static resources."""
+    assert await async_setup_component(hass, 'http', {'http': {}}) is True
+    # use the same hack as the one used in http.start()
+    # prevent freeze of http app in order to register static path
+    hass.http.app._router.freeze = lambda: None
+    client = await hass_client()
+
+    with TemporaryDirectory() as tmp_dir:
+        # test default setting is cache-able and no auth need
+        default_dir = os.path.join(tmp_dir, 'default')
+        os.mkdir(default_dir)
+
+        test_file_path = os.path.join(default_dir, 'test.jpg')
+        with open(test_file_path, 'w') as tmp_file:
```

Comment:
Do we really need to use real files, or could we patch something inside static resource to make sure the correct ting is called? Most cloud machines have slow I/O

Reply:
Just tried. Need patch 6 functions to get test working, I thought it is too much.
I can change the test to use only two new files, and they suppose are created in /tmp filesystem, is it acceptable?

Reply:
k